_Group Projects/Group_5/Week8_Assignment/Group5_Week8_Simulation.do

```diff
@@ -0,0 +1,131 @@
+bysort charter_id (tempvar): replace lottery_rank = _n // Sort tempvar in ascending order for applicants to oversubscribed schools, and then fill in lottery rank
+drop tempvar // Temporary variable no longer needed
+
+*Admitted to charter
+gen admitted = .
+replace admitted = 1 if lottery_rank<=capacity // =1 when child's lottery rank is less than school capacity
+replace admitted = 1 if oversubscribed==0 // =1 when no lottery occured
+replace admitted = 0 if oversubscribed==1 & lottery_rank>capacity // =0 for lottery losers
+la var admitted "=1 if child was admitted to charter"
+
+*Test score at charter school
+gen ts_charter = .
+replace ts_charter = iq^0.5 * charter_quality^0.5 if admitted==1 // Cobb-Douglas education production function, where a = 0.5; we chose the number based on the education production function used by Polachek, Kniesner, and Harwood (1978) in the Journal of Educational Statistics
+la var ts_charter "Child's test score if he/she attends charter"
+
+*Test score at public school
+gen ts_public = .
+replace ts_public = iq^0.5 * public_quality^0.5 // Cobb-Douglas education production function, where a = 0.5; see justification for number above
+la var ts_public "Child's test score if he/she attends public school"
+
+*School attended
+gen attend = .
+replace attend = 1 if admitted==1 & ts_charter>=ts_public // =1 if child was admitted to charter and would achieve higher test score attending charter; recall hypothetical test score at charter was generated above using Cobb-Douglas education production function
+replace attend = 0 if admitted==0 | ts_charter<ts_public // =0 if child lost charter lottery or would achieve higher test score attending public
+la var attend "=1 if child attends charter"
+
+*Actual test score
+gen ts_actual = .
+replace ts_actual = ts_charter if attend==1 // Actual test score is the hypothetical test score at charter if child attends charter
+replace ts_actual = ts_public if attend==0 // Actual test score is hypothetical test score at public if child attends public
+la var ts_actual "Child's actual test score"
+
+///*** Calculate statistical property ***///
+
+*IV regression
+reg ts_actual attend (admitted), r  // IV regression, where admitted is the instrument, attend is the endogenous variable, and ts_actual is our dependent variable; note full sample of data is used here
+
+*Estimated impact
+gen beta_hat = r(table)[1,1] // Extracts coefficient from regression
+la var beta_hat "Mean predicted change in child's test score from attending charter, relative to if child had attended public" // The latter is a hypothetical score we generated but do NOT utilize here
+
+*Estimated impact: 95% confidence intervals
+gen beta_hat_c0 = r(table)[5,1]
+la var beta_hat_c0 "Minimum of beta hat 95% CI"
+
+gen beta_hat_c1 = r(table)[6,1]
+la var beta_hat_c1 "Maximum of beta hat 95% CI"
+
+*Actual impact
+gen beta = ts_actual - ts_public if attend==1 // Note only subsample of charter attendees is used
```

Comment:
There is potentially a serious statistical issue here you will want to reconsider for your next assignment and your final assignment. Is is really the case that this is the true beta? This is the effect _on compliers among the treated_, to be sure, but it says nothing about the mean effect on treated (accepted) noncompliers, nor about the hypothetical effect on individuals who are not accepted. Which are you interested in estimating? Is this its true value?